lib/pure/hashes.nim

```diff
@@ -141,13 +141,18 @@ when defined(js):
       res = hiXorLoJs(hiXorLoJs(P0, x xor P1), P58)
     cast[Hash](toNumber(wrapToInt(res, 32)))
 
-  template asBigInt(num: float): JsBigInt =
+  template toBits(num: float): JsBigInt =
     let
       x = newArrayBuffer(8)
       y = newFloat64Array(x)
-      z = newBigUint64Array(x)
-    y[0] = num
-    z[0]
+    if hasBigUint64Array():
+      let z = newBigUint64Array(x)
+      y[0] = num
+      z[0]
+    else:
+      let z = newUint32Array(x)
+      y[0] = num
+      big(z[0]) + big(z[1]) shl big(32)
```

Comment:
```suggestion
      big(z[0]) + big(z[1]) shl big(32) # xxx support `bigEndian`
```

Reply:
`endians` is totally wrong in nodejs backend. For Nim JS, endians is always `bigEndian`.

Reply:
but we could check endians by means of typed array

```js
{.emit: """
let endianNess = () => {
  let uInt32 = new Uint32Array([0x11223344]);
  let uInt8 = new Uint8Array(uInt32.buffer);

  if(uInt8[0] === 0x44) {
      return 'Little Endian';
  } else if (uInt8[0] === 0x11) {
      return 'Big Endian';
  } else {
      return 'Maybe mixed-endian?';
  }
};

console.log(endianNess());
"""
.}
```

Reply:
> For Nim JS, endians is always bigEndian.

endianness for js (say in browser or on node) depends on the host machine where the code runs 
see also https://github.com/timotheecour/Nim/issues/515

> endians is totally wrong in nodejs backend. 

possibly

Reply:
IMO for starters we should fix these:
```nim
    (name: "js", intSize: 32, endian: bigEndian,floatSize: 64,bit: 32),
    (name: "nimvm", intSize: 32, endian: bigEndian, floatSize: 64, bit: 32),
```
to `littleEndian`
in platform.nim

Reply:
=> https://github.com/nim-lang/Nim/pull/16886

Reply:
in platform.nim
@timotheecour
there is `vm` and `nimvm`, I don't whether they are different?